src/liballoc/arc.rs

```diff
@@ -1211,8 +1360,49 @@ impl<T> From<T> for Arc<T> {
     }
 }
 
+#[unstable(feature = "shared_from_slice", issue = "40475")]
+impl<'a, T: Clone> From<&'a [T]> for Arc<[T]> {
+    #[inline]
+    default fn from(v: &[T]) -> Arc<[T]> {
+        Arc::clone_from_slice(v)
+    }
+}
+
+#[unstable(feature = "shared_from_slice", issue = "40475")]
+impl<'a, T: Copy> From<&'a [T]> for Arc<[T]> {
+    #[inline]
+    fn from(v: &[T]) -> Arc<[T]> {
+        Arc::copy_from_slice(v)
+    }
+}
+
+#[unstable(feature = "shared_from_slice", issue = "40475")]
+impl<'a> From<&'a str> for Arc<str> {
+    #[inline]
+    fn from(v: &str) -> Arc<str> {
+        unsafe { from_arc_utf8_unchecked(Arc::copy_from_slice(v.as_bytes())) }
```

Comment:
I'd personally be ok just using `transmute` here, exposing `from_arc_utf8_unchecked` is unlikely to be stabilized in the long run I think.